app/services/message_employee_matcher.rb

```diff
@@ -12,22 +12,24 @@ def run
   attr_reader :message
 
   def retrieve_matching_employees
-    Employee.where(started_on: day_count.days.ago)
-      .select { |employee| match_time(employee.time_zone) && message_not_already_sent?(employee) }
+    Employee.where(started_on: day_count.days.ago).select do |employee|
+      time_to_send_message?(employee.time_zone) && message_not_already_sent?(employee)
+    end
   end
 
   def day_count
     message.days_after_start
   end
 
-  def match_time(time_zone)
+  def time_to_send_message?(time_zone)
     employee_current_time = Time.current.in_time_zone(time_zone)
 
-    if employee_current_time.day == Time.current.day
-      employee_current_time_value = employee_current_time.strftime("%H%M").to_i
-      message_time_value = message.time_of_day.strftime("%H%M").to_i
-      employee_current_time_value >= message_time_value
-    end
+    return false if employee_current_time.day != Time.current.day
```

Comment:
I am generally not a fan of  early`return` statements in Ruby (I know they are recommended by some). I think that `if` statements are easier to understand / more readable.

In a situation like this, I would instead do something like:

```ruby
def time_to_send_message?(time_zone)
  employee_current_time = Time.current.in_time_zone(time_zone)

  if employee_current_time.day == Time.current.day
    ...
  end
end
```